slides/Asymmetric Cryptography.tex

```diff
@@ -0,0 +1,229 @@
+		\begin{itemize}
+			\item<4-> One-way function: \(Y^x\ (mod\ P)\)
+			\item<5-> Parameters: \(Y = 7\) and \(P = 11\)
+		\end{itemize}
+	\end{itemize}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Diffie-Hellman-Merkle Key Exchange}
+	\begin{table}
+		\centering
+		\resizebox{10.5cm}{!}{
+			\begin{tabular}{c c c c}
+				\hline
+				&Alice&&Bob\\
+				\hline
+				&&&\\
+				Step 1 & chooses \(A = 3\) && chooses \(B = 6\)\\
+				&&&\\
+				Step 2 & \(\alpha = Y^A\ (mod\ P)\) && \(\beta = Y^B\ (mod\ P)\) \\					& \(\alpha = 7^3\ (mod\ 11) = 2\) &&\(\beta = 7^6\ (mod\ 11) = 4\) \\
+				&&&\\
+				Step 3 & Alice sends her result \(\alpha = 2\) to Bob && Bob sends his result \(\beta = 4\) to Alice \\
+				&&&\\
+				\textit{Exchange} & \multicolumn{3}{c}{\textit{The only exchange of information (not key!)}}\\
+				&&&\\
+				Step 4 & Alice calculates \(k = \beta^A\ (mod\ P)\) && Bob calculates \(k = \alpha^B\ (mod\ P)\) \\
+				&\(k = 4^3\ (mod\ 11) = 64\ (mod\ 11) = 9\) && \(k = 2^6\ (mod\ 11) = 64\ (mod\ 11) = 9\) \\
+				&&&\\
+				\textit{Key} & \multicolumn{3}{c}{\textit{Alice and Bob have agreed on key \(k = 9\), because:}}\\
+				& \multicolumn{3}{c}{$\alpha^{B}\ (mod\ P) = Y^{AB}\ (mod\ P) = Y^{BA}\ (mod\ P) = \beta^{A}\ (mod\ P)$}\\
+				&&&\\
+				\hline
+			\end{tabular}
+			}
+		\caption{Diffie-Hellman-Merkle-Key-Exchange. Based on \cite{singh1999}}
+	\end{table}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Issues}
+	\vspace{-0.3cm}
+	\begin{figure}
+		\centering
+		\includegraphics[width = 9cm]{../assets/images/drawing_DHM_exchange.png}
+	\end{figure}
+	\vspace{0.5cm}
+	\begin{itemize}
+		\item<1-> Undermines spontaneity of online communication
+	\end{itemize}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Asymmetric Cryptography}
+	\begin{itemize}
+		\item<1-> So far: Symmetric cryptography
+		\begin{itemize}
+			\item<1->Decryption is inverse algorithm of encryption
+		\end{itemize}
+		\item<2-> Diffie's Idea (1975): Asymmetric encryption
+		\begin{itemize}
+			\item<2->Key to encrypt and key to decrypt a message are not identical
+		\end{itemize}
+		\item<3->...but how?
+	\end{itemize}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Asymmetric Cryptography}
+	\begin{itemize}
+		\item<1-> Ronald Rivest, Adi Shamir and Leonard Adleman
+		\item<2-> Inspired by Diffie's idea of asymmetric cryptography
+		\item<3-> Develop first asymmetric encryption algorithm (\textit{RSA}) \cite{rivest1978}
+	\end{itemize}
+	\uncover<3->{
+		\begin{figure}
+			\centering
+			\includegraphics[width = 8cm]{../assets/images/cover_RSA}
+		\end{figure}
+	}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{RSA}
+	Properties of RSA
+	\begin{itemize}
+		\item<1->[(a)] Alice creates a \textit{public key} with which everyone can encrypt a message. The encryption with the public key is a one-way function and therefore not revertable
+		\item<2->[(b)] Only Alice is able to decrypt the message with her \textit{private key} and revert the encryption process
+	\end{itemize}
+\end{frame}
+%%%
+
```

Comment:
What do you think about adding the two use cases "Privacy" and "Integrity/Authenticity" of asymmetric cryptography in this section? There's a nice figure in "Bitcoin, Blockchain & Cryptoassets" on that.